admin/failures/overview/endpoint.js

```diff
@@ -15,8 +15,17 @@ module.exports = (purchases) => (req, res) => {
             .sort((a, b) => {
                 let aLatestTimestamp, bLatestTimestamp;
 
-                !a.data.errors || !a.data.errors.length ? aLatestTimestamp = a.data.viewModel.date : aLatestTimestamp = a.data.errors[a.data.errors.length - 1].at;
-                !b.data.errors || !b.data.errors.length ? bLatestTimestamp = b.data.viewModel.date : bLatestTimestamp = b.data.errors[b.data.errors.length - 1].at;
+                if(!a.data.errors || !a.data.errors.length){
+                    aLatestTimestamp = a.data.viewModel.date;
+                }else{
+                    aLatestTimestamp = a.data.errors[a.data.errors.length - 1].at;
+                }
+
+                if(!b.data.errors || !b.data.errors.length){
+                    bLatestTimestamp = b.data.viewModel.date
+                }else{
+                    bLatestTimestamp = b.data.errors[b.data.errors.length - 1].at;
+                }

```

Comment:
Having a pattern like this, where you do the same thing on two items is a perfect case for extracting a function!

So `sort` becomes:

```
.sort((a, b) => {
  const aLatestTimestamp = getLatestTimestamp(a);
  const bLatestTimestamp = getLatestTimestamp(b);

  // ...
});
```